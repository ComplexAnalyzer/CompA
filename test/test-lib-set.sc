def main() 
{
    a = "hello";
    b = "hello";
    c = "hell"; 
    print(a == b);      /* prints 1 */ 
    print(a == c);      /* prints 0 */
    d = ["hello", "bye", "hi", "see", "ya"];
    e = ["hi", "see", "what"];
    f = d + e;
    (0<=i<#f) print(f[i]); /* prints hello bye hi see ya hi see what */
    print("hello"?f);      /* prints 1 */
    g = set(f);
    (0<=i<#g) print(g[i]);  /* prints hello bye hi see ya what */ 
    f = d * e;
    (0<=i<#f) print(f[i]);  /* prints hi see */
    f = d & e;
    (0<=i<#f) print(f[i]);  /* hello bye hi see ya what */
    f = d - e;
    (0<=i<#f) print(f[i]);  /* hello by ya */

}
