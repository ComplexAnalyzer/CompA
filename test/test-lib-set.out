true
false
hello
bye
hi
see
ya
hi
see
what
true
hello
bye
hi
see
ya
what
hi
see
hello
bye
hi
see
ya
what
hello
bye
ya
